samples/cellular/modem_shell/src/link/link_shell.c

```diff
@@ -2559,6 +2578,59 @@ static int link_shell_tau(const struct shell *shell, size_t argc, char **argv)
 	return 0;
 }
 
+static int link_shell_modem(const struct shell *shell, size_t argc, char **argv)
+{
+	bool operation_selected = false;
+
+	optreset = 1;
+	optind = 1;
+	int opt;
+
+	while ((opt = getopt_long(argc, argv, short_options, long_options, NULL)) != -1) {
```

Comment:
It seems that we get to fine tune command parsing again. 😄 This command expects only one option, so maybe the `while()` loop should be removed. At least this needs to be changed somehow, now it's possible to run multiple operations back to back, for example command `link modem --shutdown --init -h`, shuts down the modem, initializes it again and prints the help, which is probably not the intention.